### PR TITLE
JWT authentication, Registration, Email verification

### PR DESCRIPTION
**Pull request by shitcodestyle**
I have integrated JWT authentication, user registration with email verification. I have decided to develop JSON API so Accept and Content-type should be application/json to avoid unpredictable errors=)

**How does mail verification work**
When a user is created, a verification link is sent to the user mail. The verification link looks the following: 
   GET /auth/verifyUser?email=user@mail.com&signature=14sdasar32242
  The signature is signed user mail with secret key. The service checks the link for integrity and authority. If it is ok, the service marks a user as verified.

**JWT authentication**
JWT is not just a token going from a client to a serve. It also defines some rules which both side should stick: rules to verify a token, to grant a token, to refresh the token. JWT can be stored in HTTP cookies or HTTP authorization header. Storing it in HTTP-only cookies is more secure way as in this case JS scripts can't retrieve such cookies. Usually, a html page includes a lot of js files from unverified sources.

### CODE REVIEW

File: app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * Create a new AuthController instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+
+    }
+
+    /**
```
Comment (shitcodestyle) [inline app/Http/Controllers/AuthController.php:29]: shit code

File: app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+use App\Http\Requests\VerifyUserRequest;
+use App\Mail\VerificationMail;
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Http\JsonResponse;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Mail;
+
+
+class AuthController extends Controller
```
Comment (cybsom) [inline app/Http/Controllers/AuthController.php:17]: Вынести с контроллеров всю бизнес логику

File: app/Http/Controllers/AuthController.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        $data = $request->validated();
+        $user = new User($data);
+        $user->save();
+
+        // send verification link
+        $verification_url = route('auth.verify', ['email' => $user->email, 'signature' => $user->signed_email]);
+        Mail::to($user->email)->send(new VerificationMail($verification_url));
+
+        // response with json
+        return response()->json(['user' => $user, 'message' => "verification link was sent to email $user->email"]);
```
Comment (cybsom) [inline app/Http/Controllers/AuthController.php:63]: Унифицировать структуру ответов

File: app/Http/Requests/LoginUserRequest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+
+namespace App\Http\Requests;
+
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class LoginUserRequest extends FormRequest
```
Comment (cybsom) [inline app/Http/Requests/LoginUserRequest.php:9]: Добавь во всех классах метод getRequestDto, и в actions передавай только dto объекты

File: app/Http/Controllers/AuthController.php [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        return $this->createNewToken(auth()->refresh());
+    }
+
+    /**
+     * Get the authenticated User.
+     *
+     * @return Authenticatable
+     */
+    public function userProfile(): Authenticatable
```
Comment (cybsom) [inline app/Http/Controllers/AuthController.php:105]: Не возвращай модель, только с помощью трансформера

**Pull request merged**